### PR TITLE
[secure-mode] Add properties allowlist validation

### PR DESCRIPTION
**Pull request by ethnzhng**

### CODE REVIEW

**Comment by siddvenk** [comment]
I'd prefer a denylist instead of an allowlist With this, new configs are not allowed by default (maybe that's the desired behavior), but I feel like there are going to be new configs added that should be allowed more often than new configs added that are disallowed.

File: plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeUtils.java [hunk 1]
```diff
@@ -95,10 +95,28 @@ public static void validateSecurity(ModelInfo<?, ?> modelInfo) throws IOExceptio
...
 
+    /**
+     * In Secure Mode we only allow an explicit set of DJL-Serving properties to be set.
+     *
+     * @param modelInfo ModelInfo of the model
+     * @param allowedKeys set of allowlisted properties
+     */
+    private static void validateProperties(ModelInfo<?, ?> modelInfo, Set<String> allowedKeys) {
+        Properties prop = modelInfo.getProperties();
+        Set<String> allowedKeysSet = new HashSet<>(allowedKeys);
```
Comment (siddvenk) [inline plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeUtils.java:111]: this is redundant, allowedKeys is already a set.

File: plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeAllowList.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.plugins.securemode;
+
+import java.util.Set;
+
+/** A class for properties that are allowlisted in Secure Mode. */
+final class SecureModeAllowList {
```
Comment (frankfliu) [inline plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeAllowList.java:18]: Make this a interface

File: plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeAllowList.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    public static final Set<String> PROPERTIES_ALLOWLIST =
+            Set.of(
+                    "engine",
+                    "job_queue_size",
```
Comment (frankfliu) [inline plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeAllowList.java:23]: Add the following to whitelist:
1. max_idle_time
2. batch_size
3. max_batch_delay

Comment (sindhuvahinis) [reply plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeAllowList.java:23]: Yeah, maybe minWorkers and maxWorkers?

**Review by lanking520 — APPROVED** [review]

**Review by frankfliu — APPROVED** [review]

**Pull request merged**